keras_cv/layers/fpn/feature_pyramid.py

```diff
@@ -0,0 +1,156 @@
+            Passing None will populate the list with the default FPN operation
+            (an add operation). Defaults to None.
+
+    References:
+        [FPN paper](https://arxiv.org/pdf/1612.03144)
+
+    Sample Usage:
+    ```python
+    inp = tf.keras.layers.Input((384, 384, 3))
+    efnb0 = tf.keras.applications.EfficientNetB0(input_tensor=inp, include_top = False)
+
+    layer_names = ['block2b_add', 'block3b_add', 'block5c_add', 'top_activation']
+    backbone_outputs = [efnb0.get_layer(name).output for name in layer_names]
+
+    # features is a dict with P2, P3, P4 and P5 as keys
+    features = keras_cv.layers.FeaturePyramid([2,3,4,5])(backbone_outputs)
+    ```
+    """
+
+    def __init__(
+        self,
+        pyramid_levels,
+        num_channels=256,
+        lateral_ops=None,
+        top_down_ops=None,
+        merge_ops=None,
+        **kwargs,
+    ):
+        super(FeaturePyramid, self).__init__(**kwargs)
+        self.pyramid_levels = sorted(pyramid_levels)
+        self.num_pyramid_levels = len(self.pyramid_levels)
+        self.num_channels = num_channels
+
+        # required for successful serialization
+        self.lateral_ops_passed = lateral_ops
+        self.top_down_ops_passed = lateral_ops
+        self.merge_ops_passed = lateral_ops
+
+        if not lateral_ops:
+            # populate self.lateral_ops with default FPN Conv2D 1X1 layers
+            self.lateral_ops = []
+            for i in pyramid_levels:
+                self.lateral_ops.append(
+                    tf.keras.layers.Conv2D(
+                        self.num_channels,
+                        kernel_size=1,
+                        strides=1,
+                        padding="same",
+                        name=f"lateral_P{i}",
+                    )
+                )
+        else:
+            self.lateral_ops = lateral_ops
+
+        # the same upsampling layer is used for all levels
+        self.top_down_ops = (
+            [tf.keras.layers.UpSampling2D(size=2)] * self.num_pyramid_levels
+            if not top_down_ops
+            else top_down_ops
+        )
+        # the same merge layer is used for all levels
+        self.merge_ops = (
+            [tf.keras.layers.Add()] * (self.num_pyramid_levels - 1)
+            if not merge_ops
+            else merge_ops
+        )
+
+    def _create_pyramid(self, features):
+        # process topmost scale outside loop because no merge op required
+        output_features = {}
+        P_i = self.lateral_ops[-1](features[-1])
+        P_top_down_i = self.top_down_ops[-1](P_i)
+
+        # store the output in a dictionary
+        output_features[f"P{self.pyramid_levels[-1]}"] = P_i
+
+        # loop across all scales and perform FPN ops
+        for i in range(self.num_pyramid_levels - 2, -1, -1):
+            P_i = self.lateral_ops[i](features[i])
+            P_i = self.merge_ops[i - 1]([P_i, P_top_down_i])
+            P_top_down_i = self.top_down_ops[i](P_i)
```

Comment:
Btw, there will be a wasted top_top op, eg for a pyramid_levels [2, 3, 4, 5], there will only be 3 top drop op needed (instead of 4). The number of top down op should be same as the merge ops.